sklearn/inspection/_h_statistic.py

```diff
@@ -0,0 +1,325 @@
+    },
+    prefer_skip_nested_validation=True,
+)
+def h_statistic(
+    estimator,
+    X,
+    features=None,
+    *,
+    sample_weight=None,
+    n_max=500,
+    random_state=None,
+    eps=1e-10,
+):
+    """Friedman and Popescu's H-statistic of pairwise interaction strength.
+
+    Calculates Friedman and Popescu's H-statistic of interaction strength
+    for each feature pair j, k, see [FRI]_. The statistic is defined as:
+
+        H_jk^2 = Numerator_jk / Denominator_jk, where
+
+        - Numerator_jk = 1/n * sum(PD_{jk}(x_ij, x_ik) - PD_j(x_ij) - PD_k(x_ik)^2,
+        - Denominator_jk = 1/n * sum(PD_{jk}(x_ij, x_ik)^2),
+        - PD_j and PD_jk are the one- and two-dimensional partial dependence
+          functions centered to mean 0,
+        - and the sums run over 1 <= i <= n, where n is the sample size.
+
+    It equals the proportion of effect variability between two features unexplained
+    by their main effects. When there is no interaction, the value is
+    exactly 0. The numerator (or its square root) provides an absolute measure
+    of interaction strength, enabling direct comparison across feature pairs.
+
+    The computational complexity of the function is `O(p^2 n^2)`,
+    where `p` denotes the number of features considered. The size of `n` is
+    automatically controlled via `n_max=500`, while it is the user's responsibility
+    to select only a subset of *important* features. It is crucial to focus on important
+    features because for weak predictors, the denominator might be small, and
+    even a weak interaction could result in a high Friedman's H, sometimes exceeding 1.
+
+    Parameters
+    ----------
+    estimator : object
+        A fitted estimator.
+
+    X : {array-like or dataframe} of shape (n_samples, n_features)
+        Data for which :term:`estimator` is able to calculate predictions.
+
+    features : array-like of {int, str}, default=None
+        List of feature names or column indices used to calculate pairwise statistics.
+        The default, None, will use all column indices of X.
+
+    sample_weight : array-like of shape (n_samples,), default=None
+        Sample weights used in calculating partial dependencies.
+
+    n_max : int, default=500
+        The number of rows to draw without replacement from X (and `sample_weight`).
+
+    random_state : int, RandomState instance, default=None
+        Pseudo-random number generator used for subsampling via `n_max`.
+        See :term:`Glossary <random_state>`.
+
+    eps : float, default=1e-10
+        Threshold below which numerator values are set to 0.
+
+    Returns
+    -------
+    result : :class:`~sklearn.utils.Bunch`
+        Dictionary-like object, with the attributes listed below. Note that
+        `output_dim` equals the number of values predicted per observation.
+        For single-output regression and binary classification, `output_dim` is 1.
+
+        feature_pairs : list of length n_feature_pairs
```

Comment:
I think that it would be better to store the pair of original keys meaning if a user is passing some strings, we should store those instead of always indices.